In [1]:
from root import *

### Temperature VS Trip Duration

**Analizando el DataFrame de Weather:**

In [2]:
dfWeather.head(3)

date  max_temperature_f  mean_temperature_f  min_temperature_f  \
0 2013-08-29               74.0                68.0               61.0   
1 2013-08-30               78.0                69.0               60.0   
2 2013-08-31               71.0                64.0               57.0   

   max_dew_point_f  mean_dew_point_f  min_dew_point_f  max_humidity  \
0             61.0              58.0             56.0          93.0   
1             61.0              58.0             56.0          90.0   
2             57.0              56.0             54.0          93.0   

   mean_humidity  min_humidity   ...     mean_visibility_miles  \
0           75.0          57.0   ...                      10.0   
1           70.0          50.0   ...                      10.0   
2           75.0          57.0   ...                      10.0   

   min_visibility_miles  max_wind_Speed_mph  mean_wind_speed_mph  \
0                  10.0                23.0                 11.0   
1                   7.0                29.0                 13.0   
2                  10.0                26.0                 15.0   

   max_gust_speed_mph  precipitation_inches  cloud_cover  events  \
0                28.0                     0          4.0     NaN   
1                35.0                     0          2.0     NaN   
2                31.0                     0          4.0     NaN   

   wind_dir_degrees zip_code  
0             286.0    94107  
1             291.0    94107  
2             284.0    94107  

[3 rows x 24 columns]

In [3]:
dfWeather.isnull().sum()

date                                 0
max_temperature_f                    4
mean_temperature_f                   4
min_temperature_f                    4
max_dew_point_f                     54
mean_dew_point_f                    54
min_dew_point_f                     54
max_humidity                        54
mean_humidity                       54
min_humidity                        54
max_sea_level_pressure_inches        1
mean_sea_level_pressure_inches       1
min_sea_level_pressure_inches        1
max_visibility_miles                13
mean_visibility_miles               13
min_visibility_miles                13
max_wind_Speed_mph                   1
mean_wind_speed_mph                  1
max_gust_speed_mph                 899
precipitation_inches                 1
cloud_cover                          1
events                            3143
wind_dir_degrees                     1
zip_code                             0
dtype: int64

In [4]:
dfWeather[dfWeather.date == pd.to_datetime('2013-08-29', format='%Y-%m-%d')][['date', 'zip_code']]

date  zip_code
0   2013-08-29     94107
184 2013-08-29     94063
368 2013-08-29     94301
552 2013-08-29     94041
736 2013-08-29     95113

**Observamos que para cada fecha tenemos pronósticos de 5 zipcodes distintos:**  
Investigando, los zipcodes se corresponden de la siguiente manera

+ 94107 --> San Francisco
+ 94063 --> Redwood City
+ 94301 --> Palo Alto
+ 95113 --> San Jose

**Mientras que los zipcodes de la tabla Trips son de los usuarios que usan las bicis:**

In [5]:
print dfTrip.zip_code.unique()
print "Zip Codes distintos:", dfTrip.zip_code.unique().size

['94127' '95138' '97214' ..., '2400105' '1004004' '29609']
Zip Codes distintos: 7440


Por ejemplo:

+ 94127   --> San Francisco
+ 95138   --> San Jose
+ 97214   --> Portland
+ 2400105 --> Japon
+ 29609   --> Greenville

*Estos zip codes en principio no nos servirían para nada más que para saber de donde es la persona que realizó el viaje*

**Las ciudades en las que se encuentran las estaciones son las mismas de las que tenemos información meteorológica:**  
Esto implica que podremos vincular el dato de la duración de un viaje en particular por su start_station_id con una estación en la tabla de stations y así con una ciudad y su información meteorológica.

In [6]:
dfStation.city.unique()

array(['San Jose', 'Redwood City', 'Mountain View', 'Palo Alto',
       'San Francisco'], dtype=object)

### Ploteando :
Voy a necesitar una tabla con estos datos:

| Trip_date     | Duration      | min_temp  | max_temp
|:-------------:|:-------------:|:---------:|:--------:
| XXXX-XX-XX    | xxx           | xx °F     | xx °F

Donde Duration va a ser la media/mediana o el promedio de los viajes en ese dia.
Podria hacer un multiplot para los 5 zipcodes